dask/array/core.py

```diff
@@ -3297,6 +3349,23 @@ def auto_chunks(chunks, shape, limit, dtype, previous_chunks=None):
         return tuple(chunks)
 
 
+def _split_up_single_chunk(c, this_chunksize_tolerance, max_chunk_size, proposed):
+    # Calculate by what factor we have to split this chunk
+    m = c / proposed
+    if math.ceil(m) / m > this_chunksize_tolerance:
+        # We want to smooth things potentially if rounding up would change the result
```

Comment:
nit: Would it be possible to add type hints?

Reply:
done